packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -215,6 +222,25 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     return workInProgress.child;
   }
 
+  function updateProfiler(current, workInProgress) {
+    if (enableProfileModeMetrics) {
+      // Start render timer here and push start time onto queue
+      markActualRenderTimeStarted(workInProgress);
+
+      // Let the "complete" phase know to stop the timer,
+      // And the scheduler to record the measured time.
+      workInProgress.effectTag |= CommitProfile;
+    }
+
+    // Never bail out early for Profilers.
+    // We always want to re-measure the subtree.
```

Comment:
This seems wrong. You want to measure the subtree only if there is pending work in the subtree. We shouldn't reconcile the children unless there are new props.

Reply:
Maybe my wording is wrong?

This is not forcing children to re-render. It's just measuring the "cost" of re-rendering. If the tree bails out b'c of e.g. sCU then the "actual" render time is small, which is good.

Reply:
But why are you reconciling the children? You should only do that if the props have changed. Otherwise you should skip reconciling, which is what `bailoutOnAlreadyFinishedWork` does.

Reply:
Gotcha.

I'll take another look at this this evening, after dinner. 👍